### DISCUSSION TITLE
Issue with Facebook Pixel in Next.js - Incorrect Page URL Reporting

### DISCUSSION THREAD

**Discussion by mikesawicki23** [vercel/next.js#70263] [Help]
### Summary

Hi there!

I’m encountering an issue with Facebook Pixel integration in a Next.js application. Regardless of whether I initialize the Pixel through Google Tag Manager (GTM) or directly in the code, the Pixel consistently sends the root domain as the Page URL for all events, instead of the actual page path the user is on.

**Example:**

**Expected Page URL:** https://mywebsite.com/category/product
**Facebook Pixel sends:** https://mywebsite.com/

**Steps I’ve taken so far:**
1. Tried implementing the Pixel both via GTM and directly in the Next.js codebase.
2. Checked whether the issue occurs only on dynamic routes, but it seems to happen on all pages, static and dynamic.
3. Tested with different events, but they all report the same incorrect URL.

If anyone has encountered this issue before or has suggestions on how to fix it, I’d really appreciate your help!

**Comment by Mr-Meshky** [comment]


**Reply by mikesawicki23** [reply]
Thanks but I have an access to ChatGPT too.
